Adding filters and specifying descriptor contents

sudo-bmitch opened the pull request:
This should include changes from #9.

Signed-off-by: Brandon Mitchell <git@bmitch.net>

sudo-bmitch committed:
Updating filter example and specifying descriptor contents

Signed-off-by: Brandon Mitchell <git@bmitch.net>

sudo-bmitch commented:
@michaelb990 PTAL

sudo-bmitch commented:
I also switched to filtering example to use the generic sbom descriptor to align with the previous example output. And I also changed the annotation from `reference` to `referrers`.

jdolitsky commented:
i think parts of this and #9 are duplicated .. will let the two of you sort this out @sudo-bmitch @michaelb990

sudo-bmitch commented:
@AkihiroSuda I think this addresses some of your comments in https://github.com/opencontainers/distribution-spec/pull/335. There's probably more we should be doing, but I don't know how much that belongs in blog posts and presentations vs the spec itself.

michaelb990 commented on spec.md:
```diff
@@ -496,13 +496,17 @@ When using the `last` query parameter, the `n` parameter is OPTIONAL.
-If a query results in no referrers found, an empty manifest list MUST be returned.
 
-Upon success, the response MUST be a json body in the following format (an index / manifest list):
+Upon success, the response MUST be a json body with an Index containing a list of descriptors.
```
nit:
```suggestion
Upon success, the response MUST be a JSON body with an Index containing a list of descriptors.
```

sudo-bmitch replied:
Done, thanks!

michaelb990 commented on spec.md:
```diff
@@ -496,13 +496,17 @@ When using the `last` query parameter, the `n` parameter is OPTIONAL.
-Upon success, the response MUST be a json body in the following format (an index / manifest list):
+Upon success, the response MUST be a json body with an Index containing a list of descriptors.
+Each descriptor is of an Image or Artifact manifest in the same `<name>` namespace with a `refers` field that specifies the value of `<reference>`.
+The descriptors MUST include an `artifactType` field that is set to the value of `artifactType` for an Artifact manifest, or the configuration descriptor's `mediaType` for an Image manifest.
```
I think it is optional to include the `artifactType` in the artifact manifest now, so maybe this should say:
```suggestion
The descriptors MUST include an `artifactType` field that is set to the value of `artifactType` for an Artifact manifest if present, or the configuration descriptor's `mediaType` for an Image manifest.
```

sudo-bmitch replied:
Done, thanks!
